`cryptodev` cross-compilation fails

### Steps To Reproduce

On an `x86_64-linux` machine with nixpkgs at 05a63d86da370b0b6c3b171caf19a99da6c2f7bb:

```shell
# works
nix build '.#linuxPackages_latest.cryptodev'

# broken
nix build '.#pkgsCross.aarch64-multiplatform.linuxPackages_latest.cryptodev'
nix build '.#pkgsCross.armv7l-hf-multiplatform.linuxPackages_latest.cryptodev'
```

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/sil8nl8i6kdmgsj921dpz800445yq7km-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: SHELL=/nix/store/4xw8n979xpivdc46a9ndcvyhwgif00hz-bash-5.1-p16/bin/bash
make -C /nix/store/6v2jynvkrqfjy76bgp6wc1849x4mmzkk-linux-aarch64-unknown-linux-gnu-6.1.3-dev/lib/modules/6.1.3/build M=/build/source modules
make[1]: Entering directory '/nix/store/6v2jynvkrqfjy76bgp6wc1849x4mmzkk-linux-aarch64-unknown-linux-gnu-6.1.3-dev/lib/modules/6.1.3/build'
/nix/store/6v2jynvkrqfjy76bgp6wc1849x4mmzkk-linux-aarch64-unknown-linux-gnu-6.1.3-dev/lib/modules/6.1.3/source/Makefile:768: /nix/store/6v2jynvkrqfjy76bgp6wc1849x4mmzkk-linux-aarch64-unknown-linux-gnu-6.1.3-dev/lib/modules/6.1.3/source/ar>
make[1]: *** No rule to make target '/nix/store/6v2jynvkrqfjy76bgp6wc1849x4mmzkk-linux-aarch64-unknown-linux-gnu-6.1.3-dev/lib/modules/6.1.3/source/arch/x86/Makefile'.  Stop.
make[1]: Leaving directory '/nix/store/6v2jynvkrqfjy76bgp6wc1849x4mmzkk-linux-aarch64-unknown-linux-gnu-6.1.3-dev/lib/modules/6.1.3/build'
make: *** [Makefile:27: build] Error 2
```

### Additional context

I'm trying to bisect this, but I haven't yet found a commit where cross-compilation works.

It also fails at these commits:

- 1a7c2f41c3bf6d7fd443e2453e7b840cdc413375 (`nixos-21.05-aarch64`)

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@fortuneteller2k @NickCao 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.15, NixOS, 22.11 (Raccoon), 22.11.20221224.dac57a4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(majiir): `""`
 - channels(root): `""`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

--- a/pkgs/os-specific/linux/cryptodev/default.nix
+++ b/pkgs/os-specific/linux/cryptodev/default.nix
@@ -14,9 +14,11 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = kernel.moduleBuildDependencies;
   hardeningDisable = [ "pic" ];
 
-  KERNEL_DIR = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";
-  INSTALL_MOD_PATH = "\${out}";
-  prefix = "\${out}";
+  makeFlags = kernel.makeFlags ++ [
+    "KERNEL_DIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+    "prefix=$(out)"
+  ];
 
   meta = {
     description = "Device that allows access to Linux kernel cryptographic drivers";

